Streetcode/Streetcode.XUnitTest/MediatRTests/News/UpdateNewsTest.cs

```diff
@@ -0,0 +1,173 @@
+                It.IsAny<Expression<Func<Image, bool>>>(),
+                It.IsAny<Func<IQueryable<Image>, IIncludableQueryable<Image, object>>>()
+            )).ReturnsAsync((Image)null);
+
+            _blobService.Setup(x => x.FindFileInStorageAsBase64(It.IsAny<string>()))
+                .Returns("base64Image");
+
+            var handler = new UpdateNewsHandler(_mockRepository.Object, _mockMapper.Object, _blobService.Object);
+
+            // Act
+            var result = await handler.Handle(new UpdateNewsCommand(testNewsDTO), CancellationToken.None);
+
+            // Assert
+            Assert.Multiple(() =>
+            {
+                Assert.True(result.IsSuccess);
+                Assert.IsType<NewsDTO>(result.Value);
+            });
+        }
+
+
+
+        [Theory]
+        [InlineData(1)]
+        public async Task ShouldThrowExeption_TryMapNullRequest(int returnNuber)
+        {
+            //Arrange
+            _mockRepository.Setup(x => x.NewsRepository.Update(GetNewsWithNotExistId()!));
+            _mockRepository.Setup(x => x.SaveChangesAsync()).ReturnsAsync(returnNuber);
+
+            _mockMapper.Setup(x => x.Map<DAL.Entities.News.News>(It.IsAny<NewsDTO>()))
+                .Returns(GetNewsWithNotExistId()!);
+
+            var expectedError = "Cannot convert null to news";
+
+            var handler = new UpdateNewsHandler( _mockRepository.Object, _mockMapper.Object, _blobService.Object);
+
+            //Act
+            var result = await handler.Handle(new UpdateNewsCommand(GetNewsDTOWithNotExistId()!), CancellationToken.None);
+
+            //Assert
+            Assert.Multiple(
+                () => Assert.False(result.IsSuccess),
+                () => Assert.Equal(expectedError, result.Errors.First().Message)
+            );
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        public async Task ShouldThrowException_SaveChangesAsyncIsNotSuccessful(int returnNumber)
+        {
+            // Arrange
+            var testNews = GetNews();
+            var testNewsDTO = GetNewsDTO();
+
+            _mockRepository.Setup(x => x.NewsRepository.Update(It.IsAny<DAL.Entities.News.News>()));
+            _mockRepository.Setup(x => x.SaveChangesAsync()).ReturnsAsync(returnNumber);
+
+            _mockMapper.Setup(x => x.Map<DAL.Entities.News.News>(It.IsAny<NewsDTO>()))
+                .Returns(testNews);
+            _mockMapper.Setup(x => x.Map<NewsDTO>(It.IsAny<DAL.Entities.News.News>()))
+                .Returns(testNewsDTO);
+            _mockRepository.Setup(x => x.ImageRepository.GetFirstOrDefaultAsync(
+                    It.IsAny<Expression<Func<Image, bool>>>(),
+                    It.IsAny<Func<IQueryable<Image>, IIncludableQueryable<Image, object>>>()
+                )).ReturnsAsync((Image)null);
+
+            _blobService.Setup(x => x.FindFileInStorageAsBase64(It.IsAny<string>()))
+                .Returns("base64Image");
```

Comment:
also try to move setups to private methods, so you don't need to repeat your code.